plugins/cloud-gce/src/main/java/org/elasticsearch/cloud/gce/GceComputeService.java

```diff
@@ -20,22 +20,47 @@
 package org.elasticsearch.cloud.gce;
 
 import com.google.api.services.compute.model.Instance;
-import org.elasticsearch.common.component.AbstractLifecycleComponent;
 import org.elasticsearch.common.component.LifecycleComponent;
 
+import java.io.IOException;
 import java.util.Collection;
 
 /**
  *
  */
 public interface GceComputeService extends LifecycleComponent<GceComputeService> {
-    static final public class Fields {
+    final class Fields {
         public static final String PROJECT = "cloud.gce.project_id";
         public static final String ZONE = "cloud.gce.zone";
+        /**
+         * Default network card used when using network.host: _gce_
+         * Defaults to 0
+         */
+        public static final String NETWORK_CARD = "cloud.gce.network.card";
         public static final String REFRESH = "cloud.gce.refresh_interval";
         public static final String TAGS = "discovery.gce.tags";
         public static final String VERSION = "Elasticsearch/GceCloud/1.0";
     }
 
-    public Collection<Instance> instances();
+    /**
+     * Return a collection of running instances within the same GCE project
+     * @return a collection of running instances within the same GCE project
+     */
+    Collection<Instance> instances();
+
+    /**
+     * <p>Resolve the network address on the current running machine
+     * by asking GCE metadata instance (call to
+     * http://metadata.google.internal/computeMetadata/v1/instance/xxx).</p>
+     * <p>With xxx equals to:</p>
```

Comment:
these are just examples right? there can be anything. Maybe just call this whole method resolveMetadata(String metadataPath)?